nbgrader/apps/baseapp.py

```diff
@@ -160,6 +165,110 @@ def __init__(self, *args, **kwargs):
         super(BaseNbGraderApp, self).__init__(*args, **kwargs)
         self._nbgrader_config = NbGraderConfig(parent=self)
 
+        
+# These are the aliases and flags for nbgrader apps that inherit only from
+# TransferApp
+transfer_aliases = {}
+transfer_aliases.update(nbgrader_aliases)
+transfer_aliases.update({
+    "timezone": "TransferApp.timezone"
+})
+transfer_flags = {}
+transfer_flags.update(nbgrader_flags)
+transfer_flags.update({
+})
+        
+class TransferApp(BaseNbGraderApp):
+    """A base class for the release, collect, fetch and submit apps.
+    
+    All of these apps involve transfering files between an instructor or students
+    files and the nbgrader exchange.
+    """
+    
+    timezone = Unicode(
+        "UTC", config=True,
+        help="Timezone for recording timestamps"
+    )
+
+    timestamp_format = Unicode(
+        "%Y-%m-%d %H:%M:%S %Z", config=True,
+        help="Format string for timestamps"
+    )
+
+    def set_timestamp(self):
+        """Set the timestap."""
+        tz = gettz(self.timezone)
+        if tz is None:
+            self.fail("Invalid timezone: {}".format(self.timezone))
+        self.timestamp = datetime.datetime.now(tz).strftime(self.timestamp_format)
+    
+    ignore = List(
+        [
+            ".ipynb_checkpoints",
+            "*.pyc",
+            "__pycache__"
+        ],
+        config=True,
+        help=dedent(
+            """
+            List of file names or file globs to be ignored when copying directories.
+            """
+        )
+    )
+    
+    exchange_directory = Unicode(
+        "/srv/nbgrader/exchange",
+        config=True,
+        help="The preexisting exchange directory writable to everyone."
+    )
+
+    def ensure_exchange_directory(self):
+        """See if the exchange directory exists and is writable, fail if not."""
+        if not check_directory(self.exchange_directory, write=True, execute=True):
+            self.fail("Unwritable directory, please contact your instructor: {}".format(self.exchange_directory))
+
+    def init_args(self):
+        pass
+
+    @catch_config_error
+    def initialize(self, argv=None):
+        super(TransferApp, self).initialize(argv)
+        self.init_args()
+        self.ensure_exchange_directory()
+        self.set_timestamp()
```

Comment:
I don't think they do now, but I think I will leave it for now as I think the other commands may become timestamp aware.